dagshub/upload/wrapper.py

```diff
@@ -24,6 +27,57 @@ def get_default_branch(src_url, owner, reponame, auth):
     return res.json().get('default_branch')
 
 
+def create_repo(repo_name, is_org=False, org_name="", description="", private=False, auto_init=False, gitignores="",
+                license="", readme="",
+                template="notebook-template"):
+    import dagshub.auth
+    from dagshub.auth.token_auth import HTTPBearerAuth
+
+    token = dagshub.auth.get_token(code_input_timeout=0)
+    if token is not None:
+        auth = HTTPBearerAuth(token)
+
+    if license is None and readme is None and template is None and gitignores is None:
+        raise RuntimeError(
+            "In order to create the repository please choose a project template. "
+            "Our options are cookiecutter-dagshub-dvc or notebook-template.")
+
+    data = {
+        "name": repo_name,
+        "description": description,
+        "private": private,
+        "auto_init": auto_init,
+        "gitignores": gitignores,
+        "license": license,
+        "readme": readme,
+        "project_template": template,
+    }
+
+    url = REPO_CREATE_URL
+    if is_org is True:
+        url = ORG_REPO_CREATE_URL.format(
+            orgname=org_name,
+        )
+
+    res = requests.post(
+        urllib.parse.urljoin(config.host, url),
+        data,
+        auth=auth)
+
+    if res.status_code != HTTPStatus.CREATED:
+        logger.error(f"Response ({res.status_code}):\n"
+                     f"{res.content}")
+        if res.status_code == HTTPStatus.UNPROCESSABLE_ENTITY:
+            raise RuntimeError("Repository name is invalid or it already exists.")
+        else:
+            raise RuntimeError("Failed to create the desired repository.")
+    else:
+        logger.debug(f"Response ({res.status_code})\n")
+
+    repo = res.json()
+    return Repo(owner=repo["owner"]["login"], name=repo["name"], username=repo["owner"]["login"], token=token)
```

Comment:
The username might be an org name - not useful for auth
```suggestion
    return Repo(owner=repo["owner"]["login"], name=repo["name"], token=token)
```